.github/workflows/build_and_test.yml

```diff
@@ -0,0 +1,31 @@
+name: build_and_test
+
+on:
+  push:
+  pull_request:
+  repository_dispatch:
+    types: [backend_automation]
+  workflow_dispatch:
+
+jobs:
+  build_and_test:
+    # We want to run on external PRs, but not on our own internal PRs
+    # as they'll be run by the push to the branch.
+    if: ( github.event_name == 'push' ||
+        github.event.pull_request.head.repo.full_name != github.repository ) &&
+        ! endsWith(github.ref, '-notest')
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: build
+        run: |
+          cmake .
+          make
+      - name : test
+        run: |
+          make test
```

Comment:
Need more steps at the end: telegram message on errors and artifacts collecting:
```
      - name: call action to send Telegram message on failure
        env:
          TELEGRAM_TOKEN: ${{ secrets.TELEGRAM_CORE_TOKEN }}
          TELEGRAM_TO: ${{ secrets.TELEGRAM_CORE_TO }}
        uses: ./.github/actions/send-telegram-notify
        if: failure()
      - name: artifacts
        uses: actions/upload-artifact@v2
        if: failure()
        with:
          name: release
          retention-days: 21
          path: test/var/artifacts
```